csharp_dotnetcore/11.QnAMaker/QnAMaker/BotServices.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights;
+using Microsoft.Bot.Builder.AI.QnA;
+
+namespace AspNetCore_QnA_Bot
+{
+    /// <summary>
+    /// Class represents the bot's references to external services.
+    ///
+    /// For example, Application Insights and QnaMaker services
+    /// are kept here (singletons).  These external services are configured
+    /// using the BotConfigure class (based on the contents of your ".bot" file).
+    /// </summary>
+    [Serializable]
+    public class BotServices
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BotServices"/> class.
+        /// </summary>
+        public BotServices()
+        {
+            // Given there can be multiple Qna services used in a single bot,
+            // Qna is represented as a Dictionary.  This is also modeled in the
+            // ".bot" file since the elements are named (string).
+            QnAServices = new Dictionary<string, QnAMaker>();
```

Comment:
Remove this from the constructor. Move it to the property itself. (including the comment).